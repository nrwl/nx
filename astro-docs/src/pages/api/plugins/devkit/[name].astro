---
import StarlightPage from '@astrojs/starlight/components/StarlightPage.astro';
import { getCollection, render } from 'astro:content';

export async function getStaticPaths() {
  const docs = await getCollection(
    'devkit-docs',
    // we don't want the overview page, this is custom handled via the index.astro page
    (doc) => doc.id !== 'devkit-overview' && doc.data.docType === 'devkit'
  );


  return docs.map(doc => ({
    params: { name: doc.id.split('devkit_')[1] },
    props: {
      doc
    }
  }));
}

const { doc } = Astro.props;
const { name } = Astro.params;
if (!doc) {
  throw new Error(`DevKit doc not found, ${name}`);
}

const { Content, headings } = await render(doc);
---

<StarlightPage
  frontmatter={{
    title: doc.data.title
  }}
  headings={headings || []}
>
  <Content />
</StarlightPage>
