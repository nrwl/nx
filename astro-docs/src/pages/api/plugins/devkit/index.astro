---
import StarlightPage from '@astrojs/starlight/components/StarlightPage.astro';
import { render, getEntry } from 'astro:content';

const devkitDocs = await getEntry('devkit-docs', 'devkit-overview');

if (!devkitDocs) {
  throw new Error('DevKit documentation not found');
}

const { headings, Content } = await render(devkitDocs);
---

<StarlightPage
  frontmatter={{
    title: 'DevKit Documentation',
    description: 'Documentation for the DevKit'
  }}
  headings={headings || []}
>
  <Content />
</StarlightPage>
