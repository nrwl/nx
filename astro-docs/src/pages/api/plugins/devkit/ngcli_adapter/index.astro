---
import { getEntry, render, getCollection } from 'astro:content';
import StarlightPage from '@astrojs/starlight/components/StarlightPage.astro';

const doc = await getEntry('devkit-docs', 'ngcli_adapter-overview');

if (!doc) {
  throw new Error('angular cli adapter overview doc not found');
}

const { Content, headings } = await render(doc);
---

<StarlightPage
  frontmatter={{
    title: doc.data.title
  }}
  headings={headings || []}
>
  <Content />
</StarlightPage>
