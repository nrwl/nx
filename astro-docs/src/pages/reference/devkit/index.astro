---
import StarlightPage from '@astrojs/starlight/components/StarlightPage.astro';
import { render, getEntry } from 'astro:content';

const devkitDocs = await getEntry('nx-reference-packages', 'devkit-overview');

if (!devkitDocs) {
  throw new Error('DevKit documentation not found');
}

const { headings, Content } = await render(devkitDocs);
---

<StarlightPage
  frontmatter={{
    title: 'DevKit Documentation',
    description: 'Documentation for the DevKit'
  }}
  headings={headings || []}
>
  <Content />
</StarlightPage>
