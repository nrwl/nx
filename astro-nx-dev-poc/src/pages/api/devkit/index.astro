---
import StarlightPage from '@astrojs/starlight/components/StarlightPage.astro';
import type { MarkdownHeading } from 'astro';
import { render, getCollection } from 'astro:content';

const devkitDocs = await getCollection('devkit-docs');

if (!devkitDocs) {
  throw new Error('DevKit documentation not found');
}

const byCategory = devkitDocs.reduce((acc, doc) => {
  const category = doc.data.category || 'Uncategorized';
  if (!acc[category]) {
    acc[category] = [];
  }
  acc[category].push(doc);
  return acc;
}, {});

const headings: MarkdownHeading[] = Object.keys(byCategory).map((category) => ({
  depth: 2,
  text: category,
  slug: category.toLowerCase().replace(/\s+/g, '-'),
}));
---

<StarlightPage
  frontmatter={{
    title: 'DevKit Documentation',
    description: 'Documentation for the DevKit',
    headings: headings || [],
  }}
>
  {
    Object.entries(byCategory).map(([category, docs]) => (
      <section>
        <h2>{category}</h2>
        <ul>
          {docs.map((doc) => {
            return (
              <li>
                <a
                  href={
                    '/api/devkit/' +
                    category +
                    '#' +
                    doc.data.title.toLowerCase()
                  }
                >
                  {doc.data.title}
                </a>
              </li>
            );
          })}
        </ul>
      </section>
    ))
  }
</StarlightPage>
