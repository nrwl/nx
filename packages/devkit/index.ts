/**
 * The Nx Devkit is the underlying technology used to customize Nx to support
 * different technologies and custom use-cases. It contains many utility
 * functions for reading and writing files, updating configuration,
 * working with Abstract Syntax Trees(ASTs), and more.
 *
 * As with most things in Nx, the core of Nx Devkit is very simple.
 * It only uses language primitives and immutable objects
 * (the tree being the only exception).
 *
 * @module @nx/devkit
 */

// eslint-disable-next-line @typescript-eslint/no-restricted-imports
export * from 'nx/src/devkit-exports';
export * from './public-api';
