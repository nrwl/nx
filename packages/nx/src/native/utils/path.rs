use crate::native::{types::FileData, utils::normalize_trait::Normalize};
use std::path::{Path, PathBuf};

impl Normalize for Path {
    fn to_normalized_string(&self) -> String {
        normalize_nx_path(self)
    }
}

impl Normalize for PathBuf {
    fn to_normalized_string(&self) -> String {
        normalize_nx_path(self)
    }
}

fn normalize_nx_path<P>(path: P) -> String
where
    P: AsRef<Path>,
{
    if path.as_ref() == Path::new("") {
        return ".".into();
    }

    // convert back-slashes in Windows paths, since the js expects only forward-slash path separators
    if cfg!(windows) {
        path.as_ref().display().to_string().replace('\\', "/")
    } else {
        path.as_ref().display().to_string()
    }
}

pub fn get_child_files<P: AsRef<Path>>(directory: P, files: Vec<FileData>) -> Vec<String> {
    files
        .into_iter()
        .filter(|file_data| Path::new(&file_data.file).starts_with(directory.as_ref()))
        .map(|file_data| file_data.file)
        .collect()
}

#[cfg(test)]
mod test {
    use super::*;
    use std::path::PathBuf;

    #[test]
    fn should_get_child_files() {
        let directory = PathBuf::from("foo");
        let files = vec![
            FileData {
                file: "foo/bar".into(),
                hash: "123".into(),
            },
            FileData {
                file: "foo/baz".into(),
                hash: "123".into(),
            },
            FileData {
                file: "foo/child/bar".into(),
                hash: "123".into(),
            },
            FileData {
                file: "bar/baz".into(),
                hash: "123".into(),
            },
            FileData {
                file: "foo-other/not-child".into(),
                hash: "123".into(),
            },
        ];
        let child_files = get_child_files(&directory, files);
        assert_eq!(child_files, ["foo/bar", "foo/baz", "foo/child/bar",]);
    }
}
