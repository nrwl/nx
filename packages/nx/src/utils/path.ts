import * as path from 'path';
import { workspaceRoot } from './workspace-root';

function removeWindowsDriveLetter(osSpecificPath: string): string {
  return osSpecificPath.replace(/^[a-zA-Z]:/, '');
}

/**
 * Coverts an os specific path to a unix style path. Use this when writing paths to config files.
 * This should not be used to read files on disk because of the removal of Windows drive letters.
 */
export function normalizePath(osSpecificPath: string): string {
  return removeWindowsDriveLetter(osSpecificPath).split('\\').join('/');
}

/**
 * Normalized path fragments and joins them. Use this when writing paths to config files.
 * This should not be used to read files on disk because of the removal of Windows drive letters.
 */
export function joinPathFragments(...fragments: string[]): string {
  return normalizePath(path.join(...fragments));
}

/**
 * When running a script with the package manager (e.g. `npm run`), the package manager will
 * traverse the directory tree upwards until it finds a `package.json` and will set `process.cwd()`
 * to the folder where it found it. The actual working directory is stored in the INIT_CWD
 * environment variable (see here: https://docs.npmjs.com/cli/v9/commands/npm-run-script#description).
 *
 * @returns The path to the current working directory.
 */
export function getCwd(): string {
  return process.env.INIT_CWD?.startsWith(workspaceRoot)
    ? process.env.INIT_CWD
    : process.cwd();
}
