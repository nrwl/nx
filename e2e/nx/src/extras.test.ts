import { parseJson } from '@nx/devkit';
import {
  checkFilesExist,
  cleanupProject,
  isNotWindows,
  newProject,
  readFile,
  readJson,
  runCLI,
  uniq,
  updateFile,
  updateJson,
} from '@nx/e2e/utils';
import { join } from 'path';

describe('Extra Nx Misc Tests', () => {
  beforeAll(() => newProject({ packages: ['@nx/web', '@nx/js', '@nx/react'] }));
  afterAll(() => cleanupProject());

  describe('Output Style', () => {
    it('should stream output', async () => {
      const myapp = 'abcdefghijklmon';
      runCLI(`generate @nx/web:app apps/${myapp}`);

      updateJson(join('apps', myapp, 'project.json'), (c) => {
        c.targets['inner'] = {
          command: 'echo inner',
        };
        c.targets['echo'] = {
          executor: 'nx:run-commands',
          options: {
            commands: ['echo 1', 'echo 2', `nx inner ${myapp}`],
            parallel: false,
          },
        };
        return c;
      });

      const withPrefixes = runCLI(`echo ${myapp} --output-style=stream`).split(
        isNotWindows() ? '\n' : '\r\n'
      );
      expect(withPrefixes).toContain(`${myapp}: 1`);
      expect(withPrefixes).toContain(`${myapp}: 2`);
      expect(withPrefixes).toContain(`${myapp}: inner`);

      const noPrefixes = runCLI(
        `echo ${myapp} --output-style=stream-without-prefixes`
      );
      expect(noPrefixes).not.toContain(`${myapp}: `);
    });
  });

  describe('Nx Plugins', () => {
    it('should use plugins defined in nx.json', () => {
      const nxJson = readJson('nx.json');
      nxJson.plugins = ['./tools/plugin'];
      updateFile('nx.json', JSON.stringify(nxJson));
      updateFile('test/project.txt', 'plugin-node');
      updateFile('test2/project.txt', 'plugin-node2');
      updateFile('test2/dependencies.txt', 'plugin-node');
      updateFile(
        'tools/plugin.js',
        `
      const { readFileSync } = require('fs');
      const { dirname } = require('path');
      module.exports = {
        createNodesV2: ['**/project.txt', (configFiles) => {
          const results = [];
          for (const configFile of configFiles) {
            const name = readFileSync(configFile, 'utf8');
            results.push([configFile, {
              projects: {
                [dirname(configFile)]: {
                  name: name,
                }
              }
            }]);
          }
          
          return results;
        }],
        createDependencies: () => {
          return [
            {
              
              source: 'plugin-node',
              /**
               * The name of a {@link ProjectGraphProjectNode} that the source project depends on
               */
              target: 'plugin-node2',
            
              type: 'implicit'
            }
          ];
        }
      };
    `
      );

      runCLI('graph --file project-graph.json');
      const projectGraphJson = readJson('project-graph.json');
      expect(projectGraphJson.graph.nodes['plugin-node']).toBeDefined();
      expect(projectGraphJson.graph.nodes['plugin-node2']).toBeDefined();
      expect(projectGraphJson.graph.dependencies['plugin-node']).toContainEqual(
        {
          type: 'implicit',
          source: 'plugin-node',
          target: 'plugin-node2',
        }
      );
    });
  });

  describe('Run Commands', () => {
    const mylib = uniq('lib');
    beforeAll(() => {
      runCLI(`generate @nx/js:lib libs/${mylib}`);
    });

    it('should not override environment variables already set when setting a custom env file path', async () => {
      updateFile(
        `.env`,
        'SHARED_VAR=shared-root-value\nROOT_ONLY=root-only-value'
      );

      updateFile(
        `apps/${mylib}/.custom.env`,
        'SHARED_VAR=shared-nested-value\nNESTED_ONLY=nested-only-value'
      );

      const envFile = `apps/${mylib}/.custom.env`;
      runCLI(
        `generate @nx/workspace:run-commands echoEnvVariables --command=echo --envFile=${envFile} --project=${mylib}`
      );

      const command =
        process.platform === 'win32'
          ? `%SHARED_VAR% %ROOT_ONLY% %NESTED_ONLY%` // Windows
          : `$SHARED_VAR $ROOT_ONLY $NESTED_ONLY`;
      updateJson(join('libs', mylib, 'project.json'), (config) => {
        config.targets.echoEnvVariables.options.command += ` ${command}`;
        return config;
      });

      const result = runCLI(`run ${mylib}:echoEnvVariables`);
      expect(result).toContain('shared-root-value');
      expect(result).not.toContain('shared-nested-value');
      expect(result).toContain('root-only-value');
      expect(result).toContain('nested-only-value');
    }, 120000);

    it('should pass options', async () => {
      updateJson(join('libs', mylib, 'project.json'), (config) => {
        config.targets.echo = {
          command: 'echo --var1={args.var1}',
          options: {
            var1: 'a',
          },
        };
        return config;
      });

      const result = runCLI(`run ${mylib}:echo`, { silent: true });
      expect(result).toContain('--var1=a');
    }, 120000);

    it('should pass unknown options', async () => {
      updateJson(join('libs', mylib, 'project.json'), (config) => {
        config.targets.echo = {
          command: 'echo',
          options: {
            var1: 'a',
          },
        };
        return config;
      });

      const result = runCLI(`run ${mylib}:echo`, { silent: true });
      expect(result).toContain('--var1=a');
    }, 120000);

    it('should interpolate provided arguments', async () => {
      const echoTarget = uniq('echo');
      updateJson(join('libs', mylib, 'project.json'), (config) => {
        config.targets[echoTarget] = {
          executor: 'nx:run-commands',
          options: {
            commands: [
              'echo "Arguments:"',
              'echo "  var1: {args.var1}"',
              'echo "  var2: {args.var2}"',
              'echo "  hyphen: {args.var-hyphen}"',
              'echo "  camel: {args.varCamelCase}"',
              'echo ""',
            ],
          },
        };
        return config;
      });

      const result = runCLI(
        `run ${mylib}:${echoTarget} --var1=a --var2=b --var-hyphen=c --varCamelCase=d`
      );
      expect(result).toContain('var1: a');
      expect(result).toContain('var2: b');
      expect(result).toContain('hyphen: c');
      expect(result).toContain('camel: d');

      const resultArgs = runCLI(
        `run ${mylib}:${echoTarget} --args="--var1=a --var2=b --var-hyphen=c --varCamelCase=d"`
      );
      expect(resultArgs).toContain('var1: a');
      expect(resultArgs).toContain('var2: b');
      expect(resultArgs).toContain('hyphen: c');
      expect(resultArgs).toContain('camel: d');
    }, 120000);

    it('should fail when a process exits non-zero', async () => {
      updateJson(join('libs', mylib, 'project.json'), (config) => {
        config.targets.error = {
          executor: 'nx:run-commands',
          options: {
            command: `exit 1`,
          },
        };
        return config;
      });

      try {
        runCLI(`run ${mylib}:error`);
        fail('Should error if process errors');
      } catch (e) {
        expect(e.stderr.toString()).toContain(
          'command "exit 1" exited with non-zero status code'
        );
      }
    });

    it('run command should not break if output property is missing in options and arguments', async () => {
      updateJson(join('libs', mylib, 'project.json'), (config) => {
        config.targets.lint ??= {};
        config.targets.lint.outputs = ['{options.outputFile}'];
        return config;
      });

      expect(() =>
        runCLI(`run ${mylib}:lint --format=json`, {
          silenceError: true,
        })
      ).not.toThrow();
    }, 1000000);

    it('should handle caching output directories containing trailing slashes', async () => {
      // this test relates to https://github.com/nrwl/nx/issues/10549
      // 'cp -a /path/dir/ dest/' operates differently to 'cp -a /path/dir dest/'
      // --> which means actual build works but subsequent populate from cache (using cp -a) does not
      // --> the fix is to remove trailing slashes to ensure consistent & expected behaviour

      const mylib = uniq('lib');

      const folder = `dist/libs/${mylib}/some-folder`;

      runCLI(`generate @nx/js:lib libs/${mylib}`);

      runCLI(
        `generate @nx/workspace:run-commands build --command=echo --outputs=${folder}/ --project=${mylib}`
      );

      const commands = [
        process.platform === 'win32'
          ? `mkdir ${folder}` // Windows
          : `mkdir -p ${folder}`,
        `echo dummy > ${folder}/dummy.txt`,
      ];
      updateJson(join('libs', mylib, 'project.json'), (config) => {
        delete config.targets.build.options.command;
        config.targets.build.options = {
          ...config.targets.build.options,
          parallel: false,
          commands: commands,
        };
        return config;
      });

      // confirm that it builds correctly
      runCLI(`build ${mylib}`);
      checkFilesExist(`${folder}/dummy.txt`);

      // confirm that it populates correctly from the cache
      runCLI(`build ${mylib}`);
      checkFilesExist(`${folder}/dummy.txt`);
    }, 120000);
  });

  describe('generate --quiet', () => {
    it('should not log tree operations or install tasks', () => {
      const output = runCLI('generate @nx/react:app --quiet test-project', {
        verbose: false,
      });
      expect(output).not.toContain('CREATE');
      expect(output).not.toContain('Installed');
    });
  });

  describe('Env File', () => {
    const libName = uniq('lib');

    beforeAll(() => {
      runCLI(
        `generate @nx/js:lib libs/${libName} --bundler=none --unitTestRunner=none --no-interactive`
      );
    });

    it('should have the right env', () => {
      updateFile(
        '.env',
        `FIRSTNAME="firstname"
LASTNAME="lastname"
NX_USERNAME=$FIRSTNAME $LASTNAME`
      );
      updateJson(join('libs', libName, 'project.json'), (config) => {
        config.targets.echo = {
          command: 'echo $NX_USERNAME',
        };
        return config;
      });
      let result = runCLI(`run ${libName}:echo`);
      expect(result).toContain('firstname lastname');

      updateFile('.env', (content) => {
        content = content.replace('firstname', 'firstname2');
        content = content.replace('lastname', 'lastname2');
        return content;
      });
      result = runCLI(`run ${libName}:echo`);
      expect(result).toContain('firstname2 lastname2');
    });

    it('should work with custom env file', () => {
      updateFile(`libs/${libName}/.custom1.env`, `hello="hello1"`);
      updateFile(`libs/${libName}/.custom2.env`, `hello="hello2"`);
      updateJson(join('libs', libName, 'project.json'), (config) => {
        config.targets.hello1 = {
          command: 'echo $hello',
          options: {
            envFile: `libs/${libName}/.custom1.env`,
          },
        };
        config.targets.hello2 = {
          command: 'echo $hello',
          options: {
            envFile: `libs/${libName}/.custom2.env`,
          },
        };
        return config;
      });
      let result = runCLI(`run ${libName}:hello1`);
      expect(result).toContain('hello1');
      result = runCLI(`run ${libName}:hello2`);
      expect(result).toContain('hello2');
      result = runCLI(`run-many --target=hello1,hello2`);
      expect(result).toContain('hello1');
      expect(result).toContain('hello2');
    });
  });

  describe('task graph inputs', () => {
    const readExpandedTaskInputResponse = (): Record<
      string,
      Record<string, string[]>
    > =>
      parseJson(
        readFile('static/environment.js').match(
          /window\.expandedTaskInputsResponse\s*=\s*(.*?);/
        )[1]
      );

    const baseLib = 'lib-base-123';
    beforeAll(() => {
      runCLI(`generate @nx/js:lib libs/${baseLib}`);
    });

    it('should correctly expand default task inputs', () => {
      runCLI('graph --file=graph.html');

      const expandedInputs =
        readExpandedTaskInputResponse()[`${baseLib}:build`];

      // Executor
      expect(expandedInputs.external).toContain('npm:@nx/js');
      // Dependency of the executor package
      expect(expandedInputs.external).toContain('npm:@nx/devkit');

      // Don't include external nodes in the snapshot
      delete expandedInputs.external;
      expect(expandedInputs).toMatchSnapshot();
    });

    it('should correctly expand dependent task inputs', () => {
      const dependentLib = 'lib-dependent-123';
      runCLI(`generate @nx/js:lib libs/${dependentLib}`);

      updateJson(join('libs', baseLib, 'project.json'), (config) => {
        config.targets['build'].inputs = ['default', '^default'];
        config.implicitDependencies = [dependentLib];
        return config;
      });

      updateJson('nx.json', (json) => {
        json.namedInputs = {
          ...json.namedInputs,
          default: ['{projectRoot}/**/*'],
        };
        return json;
      });
      runCLI('graph --file=graph.html');

      const expandedInputs =
        readExpandedTaskInputResponse()[`${baseLib}:build`];

      // Executor
      expect(expandedInputs.external).toContain('npm:@nx/js');
      // Dependency of the executor package
      expect(expandedInputs.external).toContain('npm:@nx/devkit');

      // Don't include external nodes in the snapshot
      delete expandedInputs.external;
      expect(expandedInputs).toMatchSnapshot();
    });
  });
});
