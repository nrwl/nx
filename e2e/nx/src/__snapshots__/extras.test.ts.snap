// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Extra Nx Misc Tests task graph inputs should correctly expand default task inputs 1`] = `
{
  "general": [
    ".gitignore",
    "nx.json",
  ],
  "lib-base-123": [
    "libs/lib-base-123/README.md",
    "libs/lib-base-123/package.json",
    "libs/lib-base-123/project.json",
    "libs/lib-base-123/src/index.ts",
    "libs/lib-base-123/src/lib/lib-base-123.ts",
    "libs/lib-base-123/tsconfig.json",
    "libs/lib-base-123/tsconfig.lib.json",
  ],
}
`;

exports[`Extra Nx Misc Tests task graph inputs should correctly expand dependent task inputs 1`] = `
{
  "general": [
    ".gitignore",
    "nx.json",
  ],
  "lib-base-123": [
    "libs/lib-base-123/README.md",
    "libs/lib-base-123/eslint.config.cjs",
    "libs/lib-base-123/jest.config.ts",
    "libs/lib-base-123/package.json",
    "libs/lib-base-123/project.json",
    "libs/lib-base-123/src/index.ts",
    "libs/lib-base-123/src/lib/lib-base-123.spec.ts",
    "libs/lib-base-123/src/lib/lib-base-123.ts",
    "libs/lib-base-123/tsconfig.json",
    "libs/lib-base-123/tsconfig.lib.json",
    "libs/lib-base-123/tsconfig.spec.json",
  ],
  "lib-dependent-123": [
    "libs/lib-dependent-123/README.md",
    "libs/lib-dependent-123/eslint.config.cjs",
    "libs/lib-dependent-123/jest.config.ts",
    "libs/lib-dependent-123/package.json",
    "libs/lib-dependent-123/project.json",
    "libs/lib-dependent-123/src/index.ts",
    "libs/lib-dependent-123/src/lib/lib-dependent-123.spec.ts",
    "libs/lib-dependent-123/src/lib/lib-dependent-123.ts",
    "libs/lib-dependent-123/tsconfig.json",
    "libs/lib-dependent-123/tsconfig.lib.json",
    "libs/lib-dependent-123/tsconfig.spec.json",
  ],
}
`;
